src/App.tsx

```diff
@@ -15,36 +16,97 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NAME = 'name',
+  LENGTH = 'length',
+  ANY = '',
+}
+
+function getPreparedGoods(
+  goods: string[],
+  sortField: SortType,
+  isReversed: boolean,
+): string[] {
+  const preparedGoods = [...goods];
+
+  if (sortField !== SortType.ANY) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SortType.NAME:
+          return good1.localeCompare(good2);
+        case SortType.LENGTH:
+          return good1.length - good2.length;
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState<SortType>(SortType.ANY);
+  const [isReversed, setIsReversed] = useState<boolean>(false);
+
+  const visibleGoods = getPreparedGoods(goodsFromServer, sortField, isReversed);
+  const shouldShowResetButton = sortField !== SortType.ANY || isReversed;
```

Comment:
well done) 
if you use this code two times and  more, you need create constant. In this case for **sortField !== SortType.ANY** and on 32 line. But i think.. code code should work without **SorteType ANY = ''**  on 22 line.. if i right) Thank you!